src/nat/libp2p_nat_server.erl

```diff
@@ -116,7 +116,7 @@ handle_info(renew, #state{tid=TID, transport_tcp=Pid, internal_address=IntAddr,
         {error, _Reason} ->
             ok = remove_multi_addr(TID, ExtAddr0, ExtPort0),
             lager:warning("failed to renew lease for port ~p: ~p", [{IntPort, ExtPort0}, _Reason]),
-            {stop, renew_failed}
+            {stop, renew_failed, State}
```

Comment:
I would propose we retry renewing later rather than stopping here because the router's upnp/pmp server often will restart.

Reply:
Unless stopping will start a new server that will retry itself? we *know* there was a nat server available at one point.

Reply:
this is the nat server that's stopping, not the transport.  This is just to fix the crash, hopefully with the return fixed it won't spew quite so badly.  it'll restart, but I am not sure that it'll do anything without more work on the transport side.  I'm not sure if the clearing of the state that happens in the other module is enough or if some discovery process needs to be kicked back off on the transport side.